tests/test-sources/modules/performance/byte-compile-lua.nix

```diff
@@ -248,8 +248,8 @@ in
 
       extraPlugins = with pkgs.vimPlugins; [
         nvim-lspconfig
-        # Depends on plenary-nvim
-        telescope-nvim
+        # Depends on nui-nvim
+        noice-nvim
```

Comment:
I wonder if these tests are kinda fragile; if nixpkgs changes whatever plugin we're using as a test-case from `dependencies` to `propagatedBuildInputs` that'll break our test, right?

Would it be better to construct our own test-packages instead of using "real world" examples? Similar to the stub packages in the combine-plugins test.

Reply:
Although, I see something similar is done below... so wouldn't be much more effort to just move that logic up a level and share it between these tests.

Reply:
You're completely right. I left it for another PR though. This PR is for combinePlugins. But I needed to fix tests to continue. In this PR I also changed tested plugins first. It was a nightmare. Hard to find suitable plugins. And for the last several months many changes were introduced to nixpkgs, It can break anytime in the future. So I dropped it in favor of specially crafted plugins (left the commit for history though).

Reply:
> I left it for another PR though. This PR is for combinePlugins.

SGTM. I'll leave this thread open for easy reference, but it doesn't block the PR.